diff -r -U 3 src-base/minecraft/mc/Block.java src/minecraft/mc/Block.java
--- src-base/minecraft/mc/Block.java	Tue Nov 12 15:11:33 2013
+++ src/minecraft/mc/Block.java	Tue Nov 12 15:12:11 2013
@@ -297,6 +297,7 @@
         registry.func_0_VIStringObject(173, "coal_block", (new Block(Material.stone)).func_2_Block_BlockF(5.0F).func_1_Block_BlockF(10.0F).func_0_Block_BlockBlockSound(field_9_Block_L).func_1_Block_BlockString("blockCoal").func_0_Block_BlockCreativeTab(CreativeTab.buildingBlocks).func_2_Block_BlockString("coal_block"));
         registry.func_0_VIStringObject(174, "packed_ice", (new BlockPackedIce()).func_2_Block_BlockF(0.5F).func_0_Block_BlockBlockSound(field_11_Block_L).func_1_Block_BlockString("icePacked").func_2_Block_BlockString("ice_packed"));
         registry.func_0_VIStringObject(175, "double_plant", new BlockDoublePlant());
+        wtfml.WTFML.h_initBlocks(registry);
         Iterator var10 = registry.iterator();
 
         while (var10.hasNext())
diff -r -U 3 src-base/minecraft/mc/Item.java src/minecraft/mc/Item.java
--- src-base/minecraft/mc/Item.java	Tue Nov 12 15:11:37 2013
+++ src/minecraft/mc/Item.java	Tue Nov 12 15:12:17 2013
@@ -214,6 +214,7 @@
         field_2_Item_L.func_0_VIStringObject(2265, "record_ward", (new ItemRecord("ward")).func_0_ItemString("record").func_1_Item_ItemString("record_ward"));
         field_2_Item_L.func_0_VIStringObject(2266, "record_11", (new ItemRecord("11")).func_0_ItemString("record").func_1_Item_ItemString("record_11"));
         field_2_Item_L.func_0_VIStringObject(2267, "record_wait", (new ItemRecord("wait")).func_0_ItemString("record").func_1_Item_ItemString("record_wait"));
+        wtfml.WTFML.h_initItems(field_2_Item_L);
         HashSet var1 = Sets.newHashSet(new Block[] {Blocks.air, Blocks.brewing_stand, Blocks.bed, Blocks.nether_wart, Blocks.cauldron, Blocks.flower_pot, Blocks.wheat, Blocks.reeds, Blocks.cake, Blocks.skull, Blocks.piston_head, Blocks.piston_extension, Blocks.lit_redstone_ore, Blocks.powered_repeater, Blocks.pumpkin_stem, Blocks.standing_sign, Blocks.powered_comparator, Blocks.tripwire, Blocks.lit_redstone_lamp, Blocks.melon_stem, Blocks.unlit_redstone_torch, Blocks.unpowered_comparator, Blocks.redstone_wire, Blocks.wall_sign, Blocks.unpowered_repeater, Blocks.iron_door, Blocks.wooden_door});
         Iterator var2 = Block.registry.func_0_RegistryImpl_Set().iterator();
 
@@ -358,6 +359,7 @@
                     continue;
                 }
 
+                var5 = wtfml.WTFML.h_createItemBlock(var4); if(var5 == null)
                 var5 = new ItemBlock(var4);
             }
 
diff -r -U 3 src-base/minecraft/mc/Minecraft.java src/minecraft/mc/Minecraft.java
--- src-base/minecraft/mc/Minecraft.java	Tue Nov 12 15:11:38 2013
+++ src/minecraft/mc/Minecraft.java	Tue Nov 12 15:12:18 2013
@@ -416,6 +416,7 @@
     private void func_4_Minecraft_V()
     {
         this.field_1_Minecraft_List.add(this.field_15_Minecraft_L);
+        wtfml.WTFML.h_initResourcePacks(this.field_1_Minecraft_List);
     }
 
     private ByteBuffer func_0_Minecraft_ByteBufferFile(File par1File) throws IOException
diff -r -U 3 src-base/minecraft/net/minecraft/server/MinecraftServer.java src/minecraft/net/minecraft/server/MinecraftServer.java
--- src-base/minecraft/net/minecraft/server/MinecraftServer.java	Tue Nov 12 15:11:41 2013
+++ src/minecraft/net/minecraft/server/MinecraftServer.java	Tue Nov 12 15:12:26 2013
@@ -491,6 +491,7 @@
         long var1 = System.nanoTime();
         AxisAlignedBB.func_0_AxisAlignedBB_AABBPool().func_0_AABBPool_V();
         ++this.w;
+        wtfml.WTFML.h_tick(true);
 
         if (this.Q)
         {
diff -r -U 3 src-base/minecraft_server/mc/Block.java src/minecraft_server/mc/Block.java
--- src-base/minecraft_server/mc/Block.java	Tue Nov 12 15:11:42 2013
+++ src/minecraft_server/mc/Block.java	Tue Nov 12 15:12:27 2013
@@ -291,6 +291,7 @@
         registry.func_0_VIStringObject(173, "coal_block", (new Block(Material.stone)).func_2_Block_BlockF(5.0F).func_1_Block_BlockF(10.0F).func_0_Block_BlockBlockSound(field_8_Block_L).func_1_Block_BlockString("blockCoal").func_0_Block_BlockCreativeTab(CreativeTab.buildingBlocks).func_2_Block_BlockString("coal_block"));
         registry.func_0_VIStringObject(174, "packed_ice", (new BlockPackedIce()).func_2_Block_BlockF(0.5F).func_0_Block_BlockBlockSound(field_10_Block_L).func_1_Block_BlockString("icePacked").func_2_Block_BlockString("ice_packed"));
         registry.func_0_VIStringObject(175, "double_plant", new BlockDoublePlant());
+        wtfml.WTFML.h_initBlocks(registry);
         Iterator var10 = registry.iterator();
 
         while (var10.hasNext())
diff -r -U 3 src-base/minecraft_server/mc/Item.java src/minecraft_server/mc/Item.java
--- src-base/minecraft_server/mc/Item.java	Tue Nov 12 15:11:46 2013
+++ src/minecraft_server/mc/Item.java	Tue Nov 12 15:12:32 2013
@@ -212,6 +212,7 @@
         field_2_Item_L.func_0_VIStringObject(2265, "record_ward", (new ItemRecord("ward")).func_0_ItemString("record").func_1_Item_ItemString("record_ward"));
         field_2_Item_L.func_0_VIStringObject(2266, "record_11", (new ItemRecord("11")).func_0_ItemString("record").func_1_Item_ItemString("record_11"));
         field_2_Item_L.func_0_VIStringObject(2267, "record_wait", (new ItemRecord("wait")).func_0_ItemString("record").func_1_Item_ItemString("record_wait"));
+        wtfml.WTFML.h_initItems(field_2_Item_L);
         HashSet var1 = Sets.newHashSet(new Block[] {Blocks.air, Blocks.brewing_stand, Blocks.bed, Blocks.nether_wart, Blocks.cauldron, Blocks.flower_pot, Blocks.wheat, Blocks.reeds, Blocks.cake, Blocks.skull, Blocks.piston_head, Blocks.piston_extension, Blocks.lit_redstone_ore, Blocks.powered_repeater, Blocks.pumpkin_stem, Blocks.standing_sign, Blocks.powered_comparator, Blocks.tripwire, Blocks.lit_redstone_lamp, Blocks.melon_stem, Blocks.unlit_redstone_torch, Blocks.unpowered_comparator, Blocks.redstone_wire, Blocks.wall_sign, Blocks.unpowered_repeater, Blocks.iron_door, Blocks.wooden_door});
         Iterator var2 = Block.registry.func_0_RegistryImpl_Set().iterator();
 
@@ -356,6 +357,7 @@
                     continue;
                 }
 
+                var5 = wtfml.WTFML.h_createItemBlock(var4); if(var5 == null)
                 var5 = new ItemBlock(var4);
             }
 
diff -r -U 3 src-base/minecraft_server/net/minecraft/server/MinecraftServer.java src/minecraft_server/net/minecraft/server/MinecraftServer.java
--- src-base/minecraft_server/net/minecraft/server/MinecraftServer.java	Tue Nov 12 15:11:48 2013
+++ src/minecraft_server/net/minecraft/server/MinecraftServer.java	Tue Nov 12 15:20:45 2013
@@ -501,6 +501,7 @@
         long var1 = System.nanoTime();
         AxisAlignedBB.func_0_AxisAlignedBB_AABBPool().func_0_AABBPool_V();
         ++this.w;
+        wtfml.WTFML.h_tick(true);
 
         if (this.Q)
         {
@@ -737,7 +738,10 @@
 
             if (var1)
             {
-                var15.func_1_DedicatedServer_V();
+            	final DedicatedServer var15x = var15;
+            	java.awt.EventQueue.invokeAndWait(new Runnable() {public void run() {
+                var15x.func_1_DedicatedServer_V();
+            	}});
             }
 
             var15.func_49_V();
